training/src/app/chuck-noris/chuck-noris-main/chuck-norris-main.component.ts

```diff
@@ -12,31 +16,39 @@ import { ActivatedRoute } from "@angular/router";
 })
 
 export class ChuckNorrisMain implements OnInit {
-	public jokeList$: Observable<Joke[]>;
-	public joke$: Observable<Joke>;
-	public isSearchMode: boolean = false;
+	public jokeList$: Observable<Joke[]> = this.store.select(state => state.jokeList);
+	public joke$: Observable<Joke>       = this.store.select(state => state.joke);
+	public isSearchMode: boolean         = false;
 	public isVipMode: boolean;
 	public searchInput: string;
 
-	constructor(private activatedRoute: ActivatedRoute, private routingService: RoutingService, private jokeService: JokeGeneratorService) {
+	constructor(private store: Store<AppState>, private activatedRoute: ActivatedRoute,
+				private routingService: RoutingService, private jokeService: JokeGeneratorService) {
 	}
 
 	public ngOnInit(): void {
-		this.isSearchMode = this.activatedRoute.snapshot.data['isSearchMode'];
+		this.isSearchMode = this.activatedRoute.snapshot.data[ 'isSearchMode' ];
+		this.store.select(selectJoke).subscribe((joke: Joke) => {
```

Comment:
not sure what u tried to do there? if you pass it with async pipe in the template, u dont need this